### PR TITLE
Enum and union behaviours

### PR DESCRIPTION
**Pull request by scohen**
We were not generating behaviours for enum and union fields.

### CODE REVIEW

File: lib/thrift/generator/behaviour.ex [hunk 1]
```diff
@@ -65,6 +67,17 @@ defmodule Thrift.Generator.Behaviour do
     |> FileGroup.resolve(ref)
     |> typespec(file_group)
   end
+  defp typespec(%TEnum{} = enum, _) do
```
Comment (jparise) [inline lib/thrift/generator/behaviour.ex]: Is `enum` unused hre?

File: lib/thrift/generator/behaviour.ex [hunk 2]
```diff
@@ -65,6 +67,17 @@ defmodule Thrift.Generator.Behaviour do
...
+    quote do
+      pos_integer
+    end
+  end
+  defp typespec(%Union{name: name}, file_group) do
+    dest_module = FileGroup.dest_module(file_group, name)
+    quote do
+      %unquote(dest_module){}
+    end
+  end
```
Comment (pguillory) [inline lib/thrift/generator/behaviour.ex]: Do we also need one for Exception, while you're at it?

File: test/generator/behaviour_test.exs [hunk 3]
```diff
@@ -31,5 +48,7 @@ defmodule BehaviourTest do
     assert {:my_map2, 1} in behaviour_specs
     assert {:struct_param, 1} in behaviour_specs
     assert {:my_camel_cased_function, 1} in behaviour_specs
+    assert {:get_state, 0} in behaviour_specs
+    assert {:get_sor_t, 0} in behaviour_specs
```
Comment (pguillory) [inline test/generator/behaviour_test.exs]: getSOrT would be the camelcase corresponding to get_s_or_t.

File: lib/thrift/generator/behaviour.ex [hunk 4]
```diff
@@ -65,6 +67,17 @@ defmodule Thrift.Generator.Behaviour do
     |> FileGroup.resolve(ref)
     |> typespec(file_group)
   end
+  defp typespec(%TEnum{} = enum, _) do
+    quote do
+      pos_integer
```
Comment (pguillory) [inline lib/thrift/generator/behaviour.ex]: non_neg_integer according to the spec.

File: lib/thrift/generator/behaviour.ex (hunk 2 above)
Comment (scohen) [inline lib/thrift/generator/behaviour.ex]: Ok, but that'd be very weird!

**Pull request merged**